tests/test_sft_trainer.py

```diff
@@ -980,6 +980,48 @@ def test_run_chat_style_ft(dataset_path):
         assert 'Provide two rhyming words for the word "love"' in output_inference
 
 
+@pytest.mark.parametrize(
+    "data_args",
+    [
+        (
+            # sample hugging face dataset id
+            configs.DataArguments(
+                training_data_path="lhoestq/demo1",
+                data_formatter_template="### Text:{{review}} \n\n### Stars: {{star}}",
+                response_template="\n### Stars:",
+            )
+        )
+    ],
+)
+def test_run_chat_style_add_special_tokens_ft(data_args):
+    """Check if we can perform an e2e run with chat template and multi turn chat training."""
+    with tempfile.TemporaryDirectory() as tempdir:
+
+        data_args.add_special_tokens = ["<|assistant|>", "<|user|>"]
+
+        train_args = copy.deepcopy(TRAIN_ARGS)
+        train_args.output_dir = tempdir
+
+        sft_trainer.train(MODEL_ARGS, data_args, train_args)
+
+        # validate the configs
+        _validate_training(tempdir)
+        checkpoint_path = _get_checkpoint_path(tempdir)
+
+        # Load the tokenizer
+        tokenizer = transformers.AutoTokenizer.from_pretrained(checkpoint_path)
+
+        # token out of vocabulary
+        tok1 = "<|sample_not_in_vocab123|>"
+
+        # test for out of vocab token
```

Comment:
nit: `Check that random tokens are not in tokenizer`

Reply:
do we need this check though?

Reply:
Yes this is not a required check, but while testing locally I had faced that even when `"<|user|>"` was not added to vocab it was being detected as part of vocab, not sure why that happened so initially I had added that as a check to ensure its not present without it being added!
Yeah I agree `<|sample_not_in_vocab123|>` is not a good token to test! but it was a placeholder token for issues like that.
I can remove this surely, it does not seem to be required